src/opensynth/models/faraday/gaussian_mixture/gmm_utils.py

```diff
@@ -48,6 +48,28 @@ def expand_weights(data: torch.Tensor, weights: torch.Tensor) -> torch.Tensor:
     return model_input
 
 
+def prepare_data_for_training_step(
+    data: torch.Tensor,
+    vae_module: FaradayVAE,
+    sample_weights_column: Optional[str] = None,
+) -> torch.Tensor:
+
+    encoded_data = _encode_data(data, vae_module)
+
+    if sample_weights_column is not None:
+        try:
+            model_input = _expand_weights(
+                encoded_data, data[sample_weights_column]
```

Comment:
is `data` an instance of torch.Tensor? Or is this an instance of `TrainingData`?

Reply:
Actually, yes this should be type `dict[str, torch.Tensor]` (which is the same as TrainingData)